sys/riot_hdr/riot_hdr.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2017 Kaspar Schleiser <kaspar@schleiser.de>
+ *                    Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_riot_hdr
+ * @{
+ *
+ * @file
+ * @brief       RIOT header helpers and tools
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @author      Francisco Acosta <francisco.acosta@inria.fr>
+ *
+ * @}
+ */
+
+#include <string.h>
+
+#ifdef RIOT_VERSION
+#include "cpu.h"
+#include "log.h"
+#else
+#include <stdio.h>
+#define LOG_INFO(...) printf(__VA_ARGS__)
+#endif
+
+#include "riot_hdr.h"
+#include "checksum/fletcher32.h"
+
+void riot_hdr_print(riot_hdr_t *riot_hdr)
+{
+    printf("Image magic_number: 0x%08x\n", (unsigned)riot_hdr->magic_number);
+    printf("Image Version: %#x\n", (unsigned)riot_hdr->version);
+    printf("Image start address: 0x%08x\n", (unsigned)riot_hdr->start_addr);
+    printf("Image chksum: 0x%08x\n", (unsigned)riot_hdr->chksum);
+    printf("\n");
+}
+
+int riot_hdr_validate(riot_hdr_t *riot_hdr)
+{
+    if (memcmp(riot_hdr, "RIOT", 4)) {
```

Comment:
This "RIOT" should not be a magic in here but declared as as public macro. It is also already used in two files.

Reply:
Agreed.